### PR TITLE
Implement HTTPI::Auth::SSL#ciphers configuration

### PR DESCRIPTION
**Pull request by c960657**
This is an updated version of #180 by @faucct. That PR was never merged.

### CODE REVIEW

**Comment by rogerleite** [comment]
Hi Christian.
First, thanks for "reopening" this PR. 

Before we merge, could you please check the 2 errors `HTTPI::Adapter::Curb` from the tests?

Thanks,
Roger Leite

**Comment by c960657** [comment]
The Curb tests were also failing on master. I fixed this by changing the certs to [those used in Puma tests](https://github.com/puma/puma/tree/master/examples/puma/client-certs).

I don't know what exactly was wrong with the old certificates. The certs in Puma were recently upgraded in https://github.com/puma/puma/pull/2333, but the PR just mentioned that the old certs were “outdated”.

File: spec/integration/excon_spec.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require "spec_helper"
 require "integration/support/server"
 
-describe HTTPI::Adapter::HTTPClient do
+describe HTTPI::Adapter::Excon do
```
Comment (olleolleolle) [inline spec/integration/excon_spec.rb:4]: Ah, good fix!

File: spec/integration/net_http_persistent_spec.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 require "spec_helper"
 require "integration/support/server"
+require "net/http/persistent"
 
-describe HTTPI::Adapter::NetHTTP do
+describe HTTPI::Adapter::NetHTTPPersistent do
```
Comment (olleolleolle) [inline spec/integration/net_http_persistent_spec.rb:5]: ❤️

File: spec/integration/net_http_persistent_spec.rb [hunk 3]
```diff
@@ -122,6 +123,17 @@
         response = HTTPI.get(request, adapter)
         expect(response.body).to eq("get")
       end
+
+      it "works with ciphers" do
+        skip("Requires net-http-persistent 3.x") unless Net::HTTP::Persistent::VERSION.start_with? "3."
```
Comment (olleolleolle) [inline spec/integration/net_http_persistent_spec.rb:128]: Good skip!

**Review by olleolleolle — APPROVED** [review]
Thanks so much, everyone!

A focused improvement, and with a few naming fixes in the specs, thrown in.

**Comment by olleolleolle** [comment]
@c960657 Perhaps you can add a CHANGELOG.md line (under a new `## Unreleased` heading at the top)?

**Comment by olleolleolle** [comment]
Alright, all green, and now merged.  Whew!

**Comment by edmorley** [comment]
Hi - thank you for adding this :-)

I don't suppose a new `httpi` release could be published that includes this change? It's currently only on master. My understanding is that the released equivalent feature in `savron` (https://github.com/savonrb/savon/pull/794) depends on this being fixed too.

**Pull request merged**